### ISSUE TITLE
iPhone 12 Pro BLE sensitivity

### ISSUE THREAD

**Issue by wolfgang35435** [corona-warn-app/cwa-app-ios#1531]
## Avoid duplicates
* [x] Question is not already answered in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x] Question has not already been asked in another issue

## Your Question

* Source File:
* Line(s):
* Question:
After having moved from iPhone X to iPhone 12 pro, I noticed that I got many more low risk warnings and recently even a high risk warning. I checked with the BLE scanner app and found many more BLE devices in reach for the iPhone 12 Pro compared to iPhone X. Furthermore, the measured signal strength of a BLE device is significantly larger on the iPhone 12 Pro than on the iPhone X for a BLE source placed at the same distance. (see attached pictures)
Since you use the signal strength to calculate the distance, there should be a recalibration for BLE chips with different sensitivity. How is this done and has it been done for the new iPhone series ?
![iPhone 12 Pro](https://user-images.githubusercontent.com/6039339/99417373-21dd0500-28fa-11eb-8d78-362180ada6c6.jpeg)
![iPhone X](https://user-images.githubusercontent.com/6039339/99417413-2c979a00-28fa-11eb-9f46-88e1612ee298.jpeg)

**Comment by Ein-Tim** [comment]
@wolfgang35435 Just a small remark: 
All these Apps (under iOS) can not show RPIs because they are filtered out by iOS (so on OS-Level).

**Comment by wolfgang35435** [comment]
Yes, you are right. The BLE scanner output ist not filtered. But it shows many more devices than the iPhone X, which probably means that the BLE chip in the new phone is much more sensitive. Somehow, this needs to be taken into account when the distance is calculated.

**Comment by thomasaugsten** [comment]
This is already normalised by apple to have comparable attenuation between the devices

**Comment by wolfgang35435** [comment]
Doesn't seem to work properly. Since the area goes like the square of the distance, iPhone 12 users might get many more warnings (both low risk and high risk)

**Comment by thomasaugsten** [comment]
I don't you can compare the output of a BLE Scanner which shows the plain rssi with the ENF

```
The Exposure Notifications framework exposes calibrated attenuations that aim to be device-agnostic. This Bluetooth Low-Energy (BLE) signal strength attenuation, defined as transmit power (TX_power) minus received power (RSSI), can be used as a (noisy) proxy for distance.

Attenuation = TX_power - (RSSI_measured + RSSI_correction)
```

**Comment by wolfgang35435** [comment]
Right. However, how about the quality of the correction? If the BLE hardware is sufficiently different in sensitivity, the correction might become large and the error of the correction will propagate into the distance measurement. The whole question was triggered by suddenly observing a large number of warnings after switching to the new iPhone (8 within a few days). I am a careful person avoiding any unnecessary contact.

**Comment by Ein-Tim** [comment]
@wolfgang35435 
I don't think you can really tell with the usage of BLE Scanner. Apple definitely corrects only the RPIs and these are not shown by the App. If you still get so many encounters (although you only had a few contacts) you can report back here but Issues like this are always really hard to investigate because of the high Data Privacy. 
You could share your EN-Log here so I can take a look if the ENF also recorded these Encounters (I'm sure it has, but just to be safe). 


<details>
<summary>How to export Exposure Notificaiton Logs under iOS 13.7 or newer</summary>

**German:**

`Einstellungen` ➡️ `Begegnungsmitteilungen` ➡️ `Status von Begegnungsaufzeichnungen` ➡️ `Begegnungsüberprüfungen` ➡️ `Überprüfungen exportieren`

**English:**

`Settings` ➡️ `Exposure Notifications` ➡️ `Exosure Logging Status` ➡️ `Exposure Checks` ➡️ `Export Exposure Checks`
</details>

**Comment by wolfgang35435** [comment]
Attached please find my log file
[ExposureChecks-2020-11-18.json.zip](https://github.com/corona-warn-app/cwa-app-ios/files/5559412/ExposureChecks-2020-11-18.json.zip)
. Right now, I turn Exposure Logging off when I am at home. Just turning it on for a short time got me another warning.

**Comment by wolfgang35435** [comment]
Attached please find my log file. I noticed that there are several entries with Match_count =3. Does this mean I had 3 encounters with the same infected person within 24 hours?

**Comment by Ein-Tim** [comment]
@wolfgang35435 

> Does this mean I had 3 encounters with the same infected person within 24 hours?

No, inside one hash there are many DKs (the Keys which were upload by positive people) from different positive tested people. So you can't really tell when you met them (until the Encounters start to disappear (after 14 days)). 

Could you please also share a screenshot of your CWA?

**Comment by wolfgang35435** [comment]
Ok, here is my screenshot

**Comment by Ein-Tim** [comment]
@wolfgang35435 

The upload didn't work. But it's not that important, how many encounters is CWA showing at the moment?

**Comment by wolfgang35435** [comment]
Ok, I try again - this time as a PDF. There are 9 exposures, increased risk, 7 days since last exposure.

**Comment by Ein-Tim** [comment]
@wolfgang35435 
Okay, so if there are 9 encounters this is consistent with the logged information by the ENF. 
This is all what I can do for you, just one small remark:
The quote:
`9 Encounters with High Risk` doesn't mean that you met 9 different people, it could also be that you met one person 9 times on different days.

**Comment by wolfgang35435** [comment]
Ok, thanks! What I will do for next week is to carry both phones with me to see if I get consistent results.

**Comment by Ein-Tim** [comment]
@wolfgang35435 

Yes but please note that:
> neither duration nor distance of a contact is a high-precision measurement. The bluetooth low energy (BLE) beacons are read and stored only once every 5 minutes. For this reason it can well happen that a short-time contact is seen and logged on one device but not on the other, even if the two devices were always at the same location / in short distance to each other. 

(mirrored from here: https://github.com/corona-warn-app/cwa-app-ios/issues/1469#issuecomment-723476545 (@ndgendogo)

**Comment by wolfgang35435** [comment]
Right, but on average, when observed over a longer time interval, I expect to see a similar number of encounters on both phones.

**Comment by ndegendogo** [comment]
@wolfgang35435 As @Ein-Tim already mentioned. BLE is no high-precision. If your new device has better hardware than your old, it may well be that it picks up signals that were too weak for the previous.
The expectation is of course that *in general* those delta should be of low risk. 

Please be aware that cwa screen showing 9 encounters on a red card, means: you had 9 contacts, and your overall risk score is red.
It tells no more details how many of the 9 were close or distant, long or short exposures.
Also please be aware that 9 contacts yield no good statistics.

**Comment by wolfgang35435** [comment]
I observed 7 low risk encounters over the course of about a few days and then last Saturday number 8 came and turned the risk to „red“.

 Sure, there would be no problem if the 7 low risk encounters were at a distance larger than my old iPhone could detect. Unfortunately, there is no information about the distance measured at these encounters. It would be really helpful to provide that data in the app.

**Comment by ndegendogo** [comment]
@wolfgang35435 you might also be interested in [this discussion](https://github.com/corona-warn-app/cwa-backlog/issues/23#issuecomment-677674518) back in summer.
They also compared recorded contacts with two different devices, and are discussing the differences found.

> However, the main lesson here is how extremely noisy this data is, at least when looking at weak Bluetooth signals: sometimes your SE catches something, sometimes it's your iPhone 8.

**Comment by wolfgang35435** [comment]
Now I am really confused. Suddenly my CWA turned green again telling me that my last high risk encounter was more than 14 days ago.

However, when it turned red last Saturday, it said that my last encounter was 3 days ago (which was 9 days ago from today). 

Which information is correct? How can I know how long to quarantine if the app turns red?

**Comment by ndegendogo** [comment]
@wolfgang35435 

> Which information is correct?

Both
last **high risk encounter** was more than 14 days ago
**last encounter** was 3 days ago (which was 9 days ago from today) => high or low risk, whichever was later

Maybe it helps if you remember that a "red" or "green" card shows your total risk score, not an individual contact.
(And, yes, of course, a long and near exposition contributes more to this total score than a short or distant exposition).

**Comment by wolfgang35435** [comment]
From the user’s point of view, it would be essential to know the date when the high risk threshold was exceeded . Otherwise, how should we plan for our quarantine without knowing how long it should last.

I followed the recommendation to call my doctor. He told me that I cannot get a test here in Berlin unless I have symptoms, which was not the case. He asked me when my contact with high risk happened. I told him what I though was the correct date and then he said I need to stay home for 2 weeks (until next Wednesday).

Why can’t the app provide the correct date when TRL exceed the threshold?

**Comment by Ein-Tim** [comment]
@wolfgang35435 For more information See: https://github.com/corona-warn-app/cwa-documentation/issues/422

**Comment by dsarkar** [comment]
Dear @wolfgang35435,

Actually, regarding the notification of the exact date, developers are currently working on this topic, see here https://github.com/corona-warn-app/cwa-app-ios/pull/1550.

Best wishes,
DS

---
Corona-Warn-App Open Source Team

**Comment by wolfgang35435** [comment]
Ok, excellent! This feature will significantly enhance the app.

Cheers ... Wolfgang

**Comment by ndegendogo** [comment]
@dsarkar thanks for the update. You can tell your UI designers that I like the new/ upcoming messages much more than the current. It removes a lot of ambiguity. 😀

**Comment by Ein-Tim** [comment]
Somebody else reporting this: #1672

**Comment by holicodes** [comment]
Adding my info here, I've closed my duplicate #1672 now.

I think I see the same issue as @wolfgang35435 on my iPhone 12 Pro. My match count is significantly higher than my girlfriend's on her iPhone Xs, with the same behavior. Today, I have 10 low-risk matches whereas she has 5. Since I got the new phone, I also saw two peaks without any changes in my daily life that could explain them.

I started taking notes on November 19, exactly 2 weeks after setting up the new iPhone, to have more than the 14-day view of the iOS logs. Earlier than Nov 19 (also on the old phone of course), I never exceeded 3 matches but I know that we had infection numbers rising a lot during that timeframe. Here's what my notes look like:

Date | Match count
------------ | -------------
Before | never more than 3 
Nov 19 | 5
Nov 20 | 8
Nov 21 | 11
Nov 22 | 3
Nov 23-25 | 2
Nov 26 | 1
Nov 27-30 | 0
Dec 1-2 | 0
Dec 3-7 | 1
Dec 8 | 1 at 16:01, 9 at 20:02 CET
Dec 9 | 9 at 08:05, 10 at 12:06 CET
Dec 10-11 | 10

During this time, my girlfriend never had more than half of my matches and she even used public transport a couple times whereas I didn't. As a control device, my work phone (iPhone Xs) shows zero matches so I can exclude any impact from the surroundings/neighbours of our flat. Exposure log is attached.

[ExposureChecks-2020-12-11.txt](https://github.com/corona-warn-app/cwa-app-ios/files/5679749/ExposureChecks-2020-12-11.txt)

**Comment by Ein-Tim** [comment]
Thanks @holicodes 

Could you maybe also provide

- iOS Version
- CWA Version? 

Thanks!

**Comment by holicodes** [comment]
@Ein-Tim – for me, the change you mentioned in #1703 conveniently solved this issue, too. Right after the update to 1.9.1 I now have zero encounters instead of 9 green ones.

Did this solve it for you too, @wolfgang35435?

**Comment by Ein-Tim** [comment]
@wolfgang35435 Could you give us an update? Did 1.9.1 solve the problem? 
Thank you!

**Comment by wolfgang35435** [comment]
Well, after the update, I do not get any more warnings. Maybe the new calibration has reduced the sensitivity to a level which might be to low. Unfortunately, I have sold my old iPhone X. Thus, I cannot compare to iPhone 12 Pro any more.

**Comment by Ein-Tim** [comment]
Okay, I don't think the new calibration is too low, the configuration before was to sensitive IMHO...
Do you still want this Issue to stay open @wolfgang35435?

**Comment by wolfgang35435** [comment]
I think we can close the issue.

**Comment by Ein-Tim** [comment]
@thomasaugsten could you close this?
